scripts/constants.py

```diff
@@ -1,97 +1,106 @@
-    raise ValueError(f"RPC_KEY env variable is required when targeting {NETWORK}")
-RPC_URLS = {
-    "mainnet": f"https://starknet-mainnet.infura.io/v3/{os.getenv('RPC_KEY')}",
-    "testnet": f"https://starknet-goerli.infura.io/v3/{os.getenv('RPC_KEY')}",
-    "testnet2": f"https://starknet-goerli2.infura.io/v3/{os.getenv('RPC_KEY')}",
-    "devnet": "http://127.0.0.1:5050/rpc",
-    "sharingan": os.getenv("SHARINGAN_RPC_URL"),
-    "katana": "http://127.0.0.1:5050",
-    "madara": "http://127.0.0.1:9944",
-}
-RPC_CLIENT = FullNodeClient(node_url=RPC_URLS[NETWORK])
-
 
 class ChainId(Enum):
     mainnet = int.from_bytes(b"SN_MAIN", "big")
     testnet = int.from_bytes(b"SN_GOERLI", "big")
     testnet2 = int.from_bytes(b"SN_GOERLI2", "big")
-    devnet = int.from_bytes(b"SN_GOERLI", "big")
-    sharingan = int.from_bytes(b"SN_GOERLI", "big")
     katana = int.from_bytes(b"KATANA", "big")
-    madara = int.from_bytes(b"SN_GOERLI", "big")
-
 
-BUILD_DIR = Path("build")
-BUILD_DIR.mkdir(exist_ok=True, parents=True)
-SOURCE_DIR = Path("src")
-CONTRACTS = {p.stem: p for p in list(SOURCE_DIR.glob("**/*.cairo"))}
 
+NETWORKS = {
+    "mainnet": {
+        "name": "mainnet",
+        "explorer_url": "https://starkscan.co",
+        "rpc_url": f"https://starknet-mainnet.infura.io/v3/{os.getenv('INFURA_KEY')}",
+        "chain_id": ChainId.mainnet,
+    },
+    "testnet": {
+        "name": "testnet",
+        "explorer_url": "https://testnet.starkscan.co",
+        "rpc_url": f"https://starknet-goerli.infura.io/v3/{os.getenv('INFURA_KEY')}",
+        "chain_id": ChainId.testnet,
+    },
+    "testnet2": {
+        "name": "testnet2",
+        "explorer_url": "https://testnet-2.starkscan.co",
+        "rpc_url": f"https://starknet-goerli2.infura.io/v3/{os.getenv('INFURA_KEY')}",
+        "chain_id": ChainId.testnet2,
+    },
+    "devnet": {
+        "name": "devnet",
+        "explorer_url": "",
+        "rpc_url": "http://127.0.0.1:5050/rpc",
+        "chain_id": ChainId.testnet,
+    },
+    "katana": {
+        "name": "katana",
+        "explorer_url": "",
+        "rpc_url": "http://127.0.0.1:5050",
+        "chain_id": ChainId.katana,
+    },
+    "madara": {
+        "name": "madara",
+        "explorer_url": "",
+        "rpc_url": "http://127.0.0.1:9944",
+        "chain_id": ChainId.testnet,
+    },
+    "sharingan": {
+        "name": "sharingan",
+        "explorer_url": "",
+        "rpc_url": os.getenv("SHARINGAN_RPC_URL"),
+        "chain_id": ChainId.testnet,
+    },
+}
 
-ACCOUNT_ADDRESS = os.environ.get(
-    f"{NETWORK.upper()}_ACCOUNT_ADDRESS"
+NETWORK = NETWORKS[os.getenv("STARKNET_NETWORK", "devnet")]
+NETWORK["account_address"] = os.environ.get(
+    f"{NETWORK['name'].upper()}_ACCOUNT_ADDRESS"
 ) or os.environ.get("ACCOUNT_ADDRESS")
-PRIVATE_KEY = os.environ.get(f"{NETWORK.upper()}_PRIVATE_KEY") or os.environ.get(
-    "PRIVATE_KEY"
-)
+NETWORK["private_key"] = os.environ.get(
```

Comment:
might be good to add some logging here "no specific network private key set, defaulting to private key" 

this would handle the user error that might happen when someone wants to try deploying to a new network and is in an environment with a previous private key meant for a different network

also another option is to be strict and to throw an error if they arent specifically setting a private key specific to the network in an env file.

Reply:
I'd go with warning only and non breaking because it's possible to use the same params for different chains